src/optimizing/Analyze_rule.ml

```diff
@@ -317,6 +317,7 @@ and metavarcond_step1 x =
   | R.CondNestedFormula _ -> None
   | R.CondRegexp (mvar, re, const_prop) ->
       Some (MvarRegexp (mvar, re, const_prop))
+  | R.CondType _ -> None
```

Comment:
you can add a comment like
(* TODO? maybe we should extract the strings from the type constraint *)